packages/features/di/modules/CheckBookingAndDurationLimits.ts

```diff
@@ -1,19 +1,24 @@
-import type { ICheckBookingAndDurationLimitsService } from "@calcom/features/bookings/lib/handleNewBooking/checkBookingAndDurationLimits";
+
 import { CheckBookingAndDurationLimitsService } from "@calcom/features/bookings/lib/handleNewBooking/checkBookingAndDurationLimits";
 
-import { type Container, createModule } from "../di";
+import { createModule, bindModuleToClassOnToken, type ModuleLoader } from "../di";
 import { DI_TOKENS } from "../tokens";
+import { moduleLoader as checkBookingLimitsModuleLoader } from "./CheckBookingLimits";
 
 export const checkBookingAndDurationLimitsModule = createModule();
 const token = DI_TOKENS.CHECK_BOOKING_AND_DURATION_LIMITS_SERVICE;
 const moduleToken = DI_TOKENS.CHECK_BOOKING_AND_DURATION_LIMITS_SERVICE_MODULE;
-checkBookingAndDurationLimitsModule.bind(token).toClass(CheckBookingAndDurationLimitsService, {
-  checkBookingLimitsService: DI_TOKENS.CHECK_BOOKING_LIMITS_SERVICE,
-} satisfies Record<keyof ICheckBookingAndDurationLimitsService, symbol>);
-
-export const moduleLoader = {
+const loadModule = bindModuleToClassOnToken({
+  module: checkBookingAndDurationLimitsModule,
+  moduleToken,
   token,
-  loadModule: (container: Container) => {
-    container.load(moduleToken, checkBookingAndDurationLimitsModule);
+  classs: CheckBookingAndDurationLimitsService,
+  depsMap: {
+    checkBookingLimitsService: checkBookingLimitsModuleLoader,
```

Comment:
Moving to _bindModuleToClassOnToken_ ensures that dependencies (checkBookingLimitsService) are automatically loaded.